code/weapon/weapons.cpp

```diff
@@ -8020,31 +8038,49 @@ void weapon_spew_stats(WeaponSpewType type)
 	mprintf(("\n"));
 	for (auto &wi : Weapon_info)
 	{
-		if (wi.subtype != WP_LASER)
-			continue;
-		if (wi.wi_flags[Weapon::Info_Flags::Beam])
+		if (wi.subtype != WP_LASER && wi.subtype != WP_BEAM)
 			continue;
 
 		if (all_weapons || wi.wi_flags[Weapon::Info_Flags::Player_allowed])
 		{
 			mprintf(("%s\n", wi.name));
-			mprintf(("\tVelocity: %-11.0fRange: %.0f\n", wi.max_speed, wi.max_speed * wi.lifetime));
+			//Beam range is set inthe binfo and velocity isn't very relevant to them.
+			if (wi.wi_flags[Weapon::Info_Flags::Beam])
+				mprintf((",%.2f,", wi.b_info.range));
+			else
+				mprintf(("%.2f,%.2f,", wi.max_speed, wi.max_speed * wi.lifetime));
```

Comment:
This looks like you did a blind copy-paste of the code in the other section.  Look again: this isn't making a CSV, this is outputting weapon stats suitable for loadout descriptions.

Reply:
Guilty as charged, and the same with the other item. Adjustments coming shortly